core/internal/filetransfer/file_transfer_default.go

```diff
@@ -70,8 +70,11 @@ func (ft *DefaultFileTransfer) Upload(task *Task) error {
 		}
 		req.Header.Set(parts[0], parts[1])
 	}
-	if _, err := ft.client.Do(req); err != nil {
+	resp, err := ft.client.Do(req)
+	if err != nil {
 		return err
+	} else if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("file transfer: upload: failed to upload: %v", resp.Status)
 	}
```

Comment:
Since this can upload to arbitrary URLs, we need to correctly process other success HTTP statuses:

```suggestion
	} else if resp.StatusCode < 200 || resp.StatusCode > 299 {
		return fmt.Errorf("file transfer: upload: failed to upload: %v", resp.Status)
	}
```